signer/src/error.rs

```diff
@@ -159,6 +168,11 @@ pub enum Error {
     #[error("bitcoin validation error: {0}")]
     BitcoinValidation(#[from] Box<crate::bitcoin::validation::BitcoinValidationError>),
 
+    /// An error occurred while attempting to push bytes into a bitcoin
+    /// `PushBytes` type.
+    #[error("bitcoin push-bytes error: {0}")]
+    BitcoinPushBytes(#[from] PushBytesError),
```

Comment:
For foreign types, we should use `#[source]` instead of `#[from]`. One reason why is that sometimes we want two or more different error variants with the same source error, and `#[source]` allows that.

Reply:
I'm not sure that I necessarily agree (on a general level)? In this case there is no additional context to add (that I can think of) and the error variant is very generic (`BitcoinPushBytes`), so needing to `.map_err()` feel rather superfluous.

It's valid to have a `#[from]` and additional variants using the same error as `#[source]` simultaneously, which is why I feel that such a generic variant should be the `from` and other variants adding more context use `source` (as intended).

That's my thinking anyway, but It's not a hill I'm ready to die on.

Reply:
As a general rule it makes sense to default to `#[source]` and only use `#[from]` with internal types sometimes. It promotes the right kind of thinking (add context to our errors to make debugging easier). But in this specific case I cannot I cannot think of any additional context to add, nor any other places where we'll want to use the same error and add context. If that changes, we can always just change this. So I'm actually okay with this here.